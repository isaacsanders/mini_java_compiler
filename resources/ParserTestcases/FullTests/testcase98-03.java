// Test 1
class Test1 {
	public static void main(String [] args) {
		HelperTest test = new HelperTest();

		System.out.println(test.add(1, 2));
	}
}

class HelperTest {
	int num1;
	int num2;

	public int add(int one, int two) {
		num1 = one;	//set num1
		num2 = two;	//set num2
		int added = one + two;	//add together

		return added;
	}
}