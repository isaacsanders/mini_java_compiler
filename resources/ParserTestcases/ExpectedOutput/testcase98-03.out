StmtList ::= 
Type ::= ID
CallExpr ::= NEW ID LPAREN RPAREN
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
ExprList ::= Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
ExprList ::= ExprList COMMA Expr
OptExprList ::= ExprList
CallExpr ::= CallExpr DOT ID LPAREN OptExprList RPAREN
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= SYSTEMOUTPRINTLN LPAREN Expr RPAREN SEMI
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
OptExtends ::= 
ClassVarDeclList ::= 
Type ::= INT
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
Type ::= INT
ClassVarDecl ::= Type ID SEMI
ClassVarDeclList ::= ClassVarDeclList ClassVarDecl
MethodDeclList ::= 
Type ::= INT
Type ::= INT
Formal ::= Type ID
FormalList ::= Formal
Type ::= INT
Formal ::= Type ID
FormalList ::= FormalList COMMA Formal
OptFormalList ::= FormalList
StmtList ::= 
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Stmt ::= ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Type ::= INT
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
MethodDecl ::= ... 
MethodDeclList ::= MethodDeclList MethodDecl
ClassDecl ::= ...
ClassDeclList ::= ClassDeclList ClassDecl
Program ::= MainClassDecl ClassDeclList
